### ISSUE TITLE
Cannot using the whole dataset to train

### ISSUE THREAD

**Issue by LiuzhuForFun** [open-mmlab/mmdetection#2575]
Hi，this problem has confused me a long time. I constrcuted a own dataset with 3347 images, but only 996 images that are used in the phrase of training.
The train_config file is:
` // model defination is ommited
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', with_bbox=True),
    dict(type='ColorEnhance'),
    dict(type='Resize', img_scale=[(1333, 512)],
         multiscale_mode='range', keep_ratio=True),
    dict(type='RandomFlip', flip_ratio=0.5),
    #dict(type='RandomFlip', flip_ratio=0.5, direction='vertical'),
    dict(type='Normalize', **img_norm_cfg),
    dict(type='Pad', size_divisor=32),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_bboxes', 'gt_labels']),
]

test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='ColorEnhance'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=[(960,540)],
        flip=True,
        transforms=[
            dict(type='Resize', keep_ratio=True),
            dict(type='RandomFlip'),
            dict(type='Normalize', **img_norm_cfg),
            dict(type='Pad', size_divisor=32),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img']),
        ])
]
data = dict(
    imgs_per_gpu=2,
    workers_per_gpu=2,
    train=dict(
        type=dataset_type,
        ann_file=[data_root + 'train_middle/small.json'],
        img_prefix=[data_root + 'train_middle/image/'],
        pipeline=train_pipeline),
    val=dict(
        type=dataset_type,
        ann_file=[data_root + 'test/annotations_val.json'],
        img_prefix=[data_root + 'test/image/'],
        pipeline=train_pipeline),
    test=dict(
        type=dataset_type,
        ann_file=data_root + 'test50/small.json',
        img_prefix=data_root + 'test50/data_50/',
        pipeline=test_pipeline))

optimizer = dict(type='SGD', lr=0.02/4, momentum=0.9, weight_decay=0.0001)
optimizer_config = dict(grad_clip=dict(max_norm=35, norm_type=2))

lr_config = dict(
    policy='step',
    warmup='linear',
    warmup_iters=500,
    warmup_ratio=1.0 / 3,
    step=[8, 11])
checkpoint_config = dict(interval=1)

log_config = dict(
    interval=10,
    hooks=[
        dict(type='TextLoggerHook'),
  
    ])
total_epochs = 12
dist_params = dict(backend='nccl')
log_level = 'INFO'
work_dir = './work_dirs/c-X101-aug-mid'
load_from =None
resume_from = None
workflow = [('train', 1)]`
The structure of dataset is as following: different prefixes of these images
![image](https://user-images.githubusercontent.com/29727918/80798377-d6ba0980-8bd6-11ea-83cd-120e6d2f0e39.png)
The traing is following. I cannot find out this problem, the annotation file is also complete.
![image](https://user-images.githubusercontent.com/29727918/80798539-32849280-8bd7-11ea-9219-86382947a4e0.png)
Thanks for your attention!

**Comment by Yukinoyuki1** [comment]
Hello, I met the same problem. How do you solve it?

**Comment by LiuzhuForFun** [comment]
> Hello, I met the same problem. How do you solve it?
I found this mistake comes from the incorrect annotations in my training dataset. For instance, the confict of annotations such as '0b' and '0B'. Maybe another solution is to check your config file, you maybe check the batch size.

**Comment by Yukinoyuki1** [comment]
> Hello, I met the same problem. How do you solve it?
> I found this mistake comes from the incorrect annotations in my training dataset. For instance, the confict of annotations such as '0b' and '0B'. Maybe another solution is to check your config file, you maybe check the batch size.

Thank you very much. I wrote a script and found that some pictures don't have corresponding annotations at all. And I set the batch size = 2.
